designs/NuGet-RemoteFileService.md

```diff
@@ -0,0 +1,70 @@
+
+# Spec Name
+
+* Status: In Review
+* Author(s): [Rob Relyea](https://github.com/rrelyea)
+* Issue: [10151](https://github.com/NuGet/Home/issues/10151) INuGetRemoteFileService - Fetch Images and embedded licenses for codespaces and normal execution
+
+## Problem Background
+
+With Codespaces support added to NuGet client, our code needs to be able to handle all packages and communication with feeds happening on a server node. This is architected as async method calls to services that are remoted. In most cases, we also have chosen to have 1 code path, for normal execution and remoted execution. These services are still used in the normal execution, however, the service runs in the same process, as opposed to being remoted across the network.
+
+Package Icons can either come from:
+- a package feeds search service
+- (old style) or a URI to an arbitrary place on the web
+- embedded in a package
+
+Licenses can also come from a few places, including a license file embedded in a package.
+
+## Who are the customers
+
+All PM UI customers.
+
+## Goals
+
+Support remote and normal fetching of icons and licenses.
+
+## Non-Goals
+
+## Solution Overview
+
+### Fetching Files
+Defined a new interface for this service:
+
+namespace NuGet.VisualStudio.Internal.Contracts
+{
+    public interface INuGetRemoteFileService : IDisposable

```

Comment:
I'm not in favor of `Remote` in the name.  The service abstracts away where files come from.  `INuGetFileService` seems more appropriate.

Reply:
I don't think INuGetFileService is descriptive enough. We've had APIs for files before...ok with removing Remote...but want to think through.

Reply:
V2 had a IFileService which some customers want back - https://github.com/NuGet/Home/issues/3324

Reply:
INuGetRemoteFileService - current spec
INuGetFileService - @dtivel suggestion
INuGetPackageFileService - better?
INuGetEmbeddedFileService - better?

thoughts? I lean towards INuGet*FileService

Reply:
Bear in mind that this is an internal service and doesn't interfere with implementing a V2 file service separately.

Reply:
good to know. yes.
do you still prefer your suggestion over the longer (hopefully more descriptive) names?

Reply:
Talked: going with INuGetPackageFileService

Reply:
Fixed in spec and in code PR.